internal/pkg/composable/providers/kubernetes/pod.go

```diff
@@ -539,3 +542,19 @@ func hintsCheck(annotations mapstr.M, container string, prefix string, validate
 	}
 	return hints, incorrecthints
 }
+
+// removeUnnecessaryReplicaSetData removes all data from a ReplicaSet resource, except what we need to compute
+// Pod metadata. Which is just the name and owner references.
+func removeUnnecessaryReplicaSetData(obj interface{}) (interface{}, error) {
+	old, ok := obj.(*v1.ReplicaSet)
+	if !ok {
+		return nil, fmt.Errorf("obj is not a ReplicaSet")
+	}
+	transformed := v1.ReplicaSet{}
+	transformed.ObjectMeta = kubernetes.ObjectMeta{
+		Name:            old.GetName(),
+		Namespace:       old.GetNamespace(),
+		OwnerReferences: old.GetOwnerReferences(),
```

Comment:
I think that if you don't include `ResourceVersion: old.GetResourceVersion()` here then you effectively make the `opts.IsUpdated` check inside UpdateFunc of the informer to be always true